pkg/segment/search/searchaggs_test.go

```diff
@@ -0,0 +1,363 @@
+func Test_PerformMeasureAggsOnRecsSizeLimit_NonZero_GreaterThanSegments(t *testing.T) {
+	numSegements := 2
+	recsSize := 100
+	sizeLimit := 250
+	nodeResult := &structs.NodeResult{PerformAggsOnRecs: true, RecsAggsType: structs.MeasureAggsType, MeasureOperations: []*structs.MeasureAggregator{
+		{
+			MeasureCol:  "*",
+			MeasureFunc: utils.Count,
+			StrEnc:      "count(*)",
+		},
+	}}
+	aggs := &structs.QueryAggregators{Limit: sizeLimit}
+
+	for i := 0; i < numSegements; i++ {
+		recs, finalCols := getMockRecsAndFinalCols(uint64(recsSize))
+
+		resultMap := PerformAggsOnRecs(nodeResult, aggs, recs, finalCols, uint64(numSegements), true, 1)
+
+		assert.Equal(t, uint64(i+1), nodeResult.RecsAggsProcessedSegments, "expected=%v, actual=%v", uint64(numSegements), nodeResult.RecsAggsProcessedSegments)
+
+		if i == numSegements-1 {
+			assert.Equal(t, 1, len(resultMap), "expected=%v, actual=%v", 1, len(resultMap))
+			assert.True(t, resultMap["CHECK_NEXT_AGG"], "expected=%v, actual=%v", true, resultMap["CHECK_NEXT_AGG"])
+			assert.Equal(t, 1, len(recs), "expected=%v, actual=%v", 1, len(recs))
+			for _, record := range recs {
+				assert.Equal(t, fmt.Sprint(recsSize*numSegements), record["count(*)"], "expected=%v, actual=%v", recsSize*numSegements, record["count(*)"])
+			}
+		} else {
+			assert.Nil(t, resultMap, "expected=%v, actual=%v", nil, resultMap)
+		}
+	}
+}
+
+// Tests the case where the size limit is zero and the number of segments is 2.
+// Normal case where sizelimit should not be considered.
+func Test_PerformGroupByRequestAggsOnRecsSizeLimit_Zero_MultiSegment(t *testing.T) {
+	numSegements := 2
+	sizeLimit := 0
+	recsSize := 100
+	nodeResult := &structs.NodeResult{
+		PerformAggsOnRecs: true,
+		RecsAggsType:      structs.GroupByType,
+		GroupByCols:       []string{"measure3"},
+		GroupByRequest: &structs.GroupByRequest{
+			MeasureOperations: []*structs.MeasureAggregator{
+				{
+					MeasureCol:  "*",
+					MeasureFunc: utils.Count,
+					StrEnc:      "count(*)",
+				},
+			},
+			GroupByColumns: []string{"measure3"},
+			BucketCount:    3000,
+		},
+	}
+	aggs := &structs.QueryAggregators{Limit: sizeLimit}
+
+	for i := 0; i < numSegements; i++ {
+		recs, finalCols := getMockRecsAndFinalCols(uint64(recsSize))
+
+		resultMap := PerformAggsOnRecs(nodeResult, aggs, recs, finalCols, uint64(numSegements), true, 1)
+
+		assert.Equal(t, uint64(i+1), nodeResult.RecsAggsProcessedSegments, "expected=%v, actual=%v", i+1, nodeResult.RecsAggsProcessedSegments)
+
+		if i == numSegements-1 {
+			assert.Equal(t, 1, len(resultMap), "expected=%v, actual=%v", 1, len(resultMap))
+			assert.True(t, resultMap["CHECK_NEXT_AGG"], "expected=%v, actual=%v", true, resultMap["CHECK_NEXT_AGG"])
+			assert.Equal(t, recsSize, len(recs), "expected=%v, actual=%v", recsSize, len(recs))
+			for _, record := range recs {
+				assert.Equal(t, uint64(2), record["count(*)"], "expected=%v, actual=%v", recsSize*numSegements, record["count(*)"])
```

Comment:
The two expected values (`uint64(2)` and `recsSize*numSegements`) are different

Reply:
Yes, I will update.